src/firebolt/async_db/connection.py

```diff
@@ -65,17 +69,43 @@ async def _resolve_engine_url(
             raise InterfaceError(f"Unable to retrieve engine endpoint: {e}.")
 
 
+async def _get_database_default_engine_url(
+    database: str,
+    auth: AuthTypes,
+    api_endpoint: str,
+    account_name: Optional[str] = None,
+) -> str:
+    async with AsyncClient(
+        auth=auth,
+        base_url=api_endpoint,
+        account_name=account_name,
+        api_endpoint=api_endpoint,
+    ) as client:
+        try:
+            account_id = await client.account_id
+            response = await client.get(
+                url=ACCOUNT_ENGINE_URL_BY_DATABASE_NAME.format(account_id=account_id),
+                params={"database_name": database},
+            )
+            response.raise_for_status()
+            return response.json()["engine_url"]
+        except (
+            JSONDecodeError,
+            RequestError,
+            RuntimeError,
+            HTTPStatusError,
+            KeyError,
+        ) as e:
+            raise InterfaceError(f"Unable to retrieve default engine endpoint: {e}.")
```

Comment:
What if the real error is wrong credentials or expired/incorrect token?

Reply:
Sure, we could also catch `AuthenticationError`

Reply:
On the other side, I think `AuthenticationError` is pretty representative itself, so I think we should just let it propagate further